src/app/api/newspaper/[title_id]/route.ts

```diff
@@ -0,0 +1,63 @@
+import {NextRequest, NextResponse} from 'next/server';
+import prisma from '@/lib/prisma';
+import {newspaper} from '@prisma/client';
+import {createCatalogNewspaperDtoFromIssue} from '@/models/CatalogNewspaperDto';
+import {postItemToCatalog, postMissingItemToCatalog} from '@/services/catalog.data';
+import {createCatalogMissingNewspaperDtoFromIssue} from '@/models/CatalogMissingNewspaperDto';
+
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+interface IdParams { params: { title_id: string} }
+
+// GET api/newspaper/[title_id]?box=[box]
+export async function GET(req: NextRequest, params: IdParams): Promise<NextResponse> {
+  const box = req.nextUrl.searchParams.get('box');
+  if (!box) return NextResponse.json({error: 'Box not provided'}, {status: 400});
+
+  return await prisma.newspaper.findMany({
+    where: {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      title_id: +params.params.title_id,
+      box
+    },
+    orderBy: { date: 'desc' }
+  })
```

Comment:
Ser her at params hentes både fra NextRequest-objektet og fra egen params parameter, men kunne ikke begge vært hentet fra NextRequest?

Reply:
Jo, men da blir det "const id = req.nextUrl.pathname.split('/').at(-1)". Sikkert like greit å gjøre det sånn🤔

Reply:
Eller denne er kanskje litt bedre `req.url.split('/').at(-1)`

Reply:
Så `const id = req.nextUrl.searchParams.get('title_id');` funker ikke? 🤔

Reply:
Nope, searchParams gir bare queryParams (eks. `URLSearchParams { 'box' => 'box123' }` for get newspapers

Reply:
I [docs](https://nextjs.org/docs/app/building-your-application/routing/route-handlers#dynamic-route-segments) gjør de det slik jeg hadde gjort først

Reply:
Okidoki, da bruker vi den! :)